src/main/java/com/basho/riak/client/core/operations/ts/FetchOperation.java

```diff
@@ -0,0 +1,123 @@
+package com.basho.riak.client.core.operations.ts;
+
+import com.basho.riak.client.core.converters.TimeSeriesPBConverter;
+import com.basho.riak.client.core.operations.PBFutureOperation;
+import com.basho.riak.client.core.query.timeseries.Cell;
+import com.basho.riak.client.core.query.timeseries.QueryResult;
+import com.basho.riak.client.core.util.BinaryValue;
+import com.basho.riak.protobuf.RiakKvPB;
+import com.basho.riak.protobuf.RiakMessageCodes;
+import com.google.protobuf.ByteString;
+
+import java.util.List;
+
+/**
+ * An operation to fetch a single row in a Riak Time Series table.
+ *
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.0.3
+ */
+public class FetchOperation extends PBFutureOperation<QueryResult, RiakKvPB.TsGetResp, BinaryValue>
+{
+    private final Builder builder;
+    private BinaryValue queryInfoMessage;
+
+    private FetchOperation(Builder builder)
+    {
+        super(RiakMessageCodes.MSG_TsGetReq,
+              RiakMessageCodes.MSG_TsGetResp,
+              builder.reqBuilder,
+              RiakKvPB.TsGetResp.PARSER);
+
+        this.builder = builder;
+    }
+
+    @Override
+    protected QueryResult convert(List<RiakKvPB.TsGetResp> responses)
+    {
+        // This is not a streaming op, there will only be one response
+        checkIfMoreThanOneResponse(responses);
```

Comment:
For that case, we may refactor "checkIfMoreThanOneResponse" and rename it accordingly: "checkAndGetTheSingleResponse()"